my-app/src/app/add-course-page/add-course-form/add-course-form.component.ts

```diff
@@ -9,27 +13,127 @@ import { Course } from 'src/app/models';
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 
-export class AddCourseFormComponent implements AfterContentChecked {
+export class AddCourseFormComponent implements OnInit, OnChanges {
   @Input() course: Course;
+  @Input() authorsList: ListOption[];
 
   @Output() newCourse = new EventEmitter<Partial<Course>>();
   @Output() cancel = new EventEmitter<void>();
 
   public value: Partial<Course> = {};
-  public coursePageTitle: string = "Add New course";
+  public coursePageTitle = 'Add New course';
+  public form: FormGroup;
+  public isStart = true;
+  public isValidForm = false;
 
-  ngAfterContentChecked(): void {
-    if (this.course) {
-      this.value = this.course;
-      this.coursePageTitle = "Edit course";
-    };
+  public messages = {
+    title: '',
+    description: '',
+    date: '',
+    duration: '',
+    authors: ''
+  };
+
+  get title(): AbstractControl {
+    return this.form.get('title');
+  }
+
+  get description(): AbstractControl {
+    return this.form.get('description');
+  }
+
+  get date(): AbstractControl {
+    return this.form.get('date');
+  }
+
+  get duration(): AbstractControl {
+    return this.form.get('duration');
+  }
+
+  get authors(): FormArray {
+    return this.form.get('authors') as FormArray;
+  }
+
+  constructor(private fb: FormBuilder) {}
+
+  public messagesMapped = GlobalConstants.messages;
+
+  public ngOnInit(): void {
+    this.form = this.fb.group({
+      title: ['', {validators: [Validators.required, Validators.maxLength(50)], updateOn: 'blur'}],
+      description: ['', {validators: [Validators.required, Validators.maxLength(500)], updateOn: 'blur'}],
+      date: new FormControl(),
+      duration: new FormControl(),
+      authors: new FormControl(),
+    });
   }
 
-  onSave(): void {
-    this.newCourse.emit(this.value);
+  ngOnChanges(): void {
+    if (this.course && this.isStart) {
+      this.title.setValue(this.course.name);
+      this.description.setValue(this.course.description);
+      this.date.setValue(this.course.date);
+      this.duration.setValue(this.course.length);
+      this.authors.setValue(this.course.authors);
+      this.coursePageTitle = 'Edit course';
+      this.isStart = false;
+    }
   }
 
-  onCancel(): void {
+  public onClick(controlName: string): void {
+    switch (controlName) {
+      case 'date':
+        this.setMessage(this.date, 'date');
+        break;
+      case 'duration':
+        this.setMessage(this.duration, 'duration');
+        break;
+      case 'authors':
+        this.setMessage(this.authors, 'authors');
+        break;
+  }
+  }
+
+  public onBlur(event: FocusEvent): void {
+    if ((event.target as Element).id === 'title') {
```

Comment:
define different methods for two events
do not rely on DOM and it's methods/properties, it's discouraged by angular